meson.build

```
@@ -22,6 +22,54 @@ foreach arg : global_args_maybe
 endforeach
 add_project_arguments(global_args, language: 'c')
 
+# Compute the relative path from build_root to source_root
+
+source_root = meson.current_source_dir().split('/')
+build_root = meson.global_build_root().split('/')
+
+slen = source_root.length()
+blen = build_root.length()
+
+shorter_path_len = source_root.length() >= build_root.length() ? build_root.length() : source_root.length()
+
+relative_source_parts = []
+common_prefix_len = 0
+
+foreach _ : build_root
+  if shorter_path_len > common_prefix_len and source_root[common_prefix_len] == build_root[common_prefix_len]
+    common_prefix_len += 1
+  else
+    break
+  endif
+endforeach
+
+i = 0
+foreach _ : build_root
+  if i >= common_prefix_len
+    relative_source_parts += '..'
+  endif
+  i += 1
+endforeach
+
+i = 0
+foreach dir : source_root
+  if i >= common_prefix_len
+    relative_source_parts += dir
+  endif
+  i += 1
+endforeach
```

I think sway-like implementation is simple, but what do you think?
```suggestion
i = 0
in_prefix = true
foreach p : build_root
  if not in_prefix or p != source_root[i]
    in_prefix = false
    relative_dir_parts += '..'
  endif
  i += 1
endforeach

i = 0
in_prefix = true
foreach p : source_root
  if not in_prefix or build_root[i] != p
    in_prefix = false
    relative_dir_parts += p
  endif
  i += 1
endforeach
```

---

Yup I agree, let me take it in... Thanks!!